SmartDeviceLink/SDLIAPTransport.m

```diff
@@ -249,137 +282,133 @@ - (void)connect {
     [self sdl_connect:nil];
 }
 
+/**
+ *  Cleans up after a disconnected accessory by closing any open I/O streams.
+ */
+- (void)disconnect {
+    // Stop event listening here so that even if the transport is disconnected by `SDLProxy` when there is a start session timeout, the class unregisters for accessory notifications
+    [self sdl_stopEventListening];
+
+    self.retryCounter = 0;
+    self.sessionSetupInProgress = NO;
+    self.transportDisconnected = YES;
+
+    [self.controlSession destroySession];
+    [self.dataSession destroySession];
+}
+
+
+#pragma mark Helpers
+
 /**
  *  Starts the process to connect to an accessory. If no accessory specified, scans for a valid accessory.
  *
  *  @param accessory The accessory to attempt connection with or nil to scan for accessories.
  */
 - (void)sdl_connect:(nullable EAAccessory *)accessory {
-    if (!self.session && !self.sessionSetupInProgress) {
-        // We don't have a session are not attempting to set one up, attempt to connect
-        SDLLogV(@"Session not setup, starting setup");
+    if (self.transportDisconnected) {
+        SDLLogV(@"Will not attempt to connect to an accessory because the data session disconnected. Waiting for lifecycle manager to create a new tranport object.");
```

Comment:
```suggestion
        SDLLogV(@"Will not attempt to connect to an accessory because the data session disconnected. Waiting for lifecycle manager to create a new transport object.");
```